receiver/prometheusreceiver/internal/transaction_test.go

```diff
@@ -189,6 +190,78 @@ func TestTransactionAppendSummaryNoQuantile(t *testing.T) {
 	require.ErrorIs(t, err, errEmptyQuantileLabel)
 }
 
+func TestAppendExemplarWithNoMetricName(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	labels := labels.FromStrings(
+		model.InstanceLabel, "0.0.0.0:8855",
+		model.JobLabel, "test",
+	)
+
+	_, err := tr.AppendExemplar(0, labels, exemplar.Exemplar{Value: 0})
+	assert.Equal(t, errMetricNameNotFound, err)
+}
+
+func TestAppendExemplarWithEmptyMetricName(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	labels := labels.FromStrings(
+		model.InstanceLabel, "0.0.0.0:8855",
+		model.JobLabel, "test",
+		model.MetricNameLabel, "",
+	)
+	_, err := tr.AppendExemplar(0, labels, exemplar.Exemplar{Value: 0})
+	assert.Equal(t, errMetricNameNotFound, err)
+}
+
+func TestAppendExemplarWithDuplicateLabels(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	labels := labels.FromStrings(
+		model.InstanceLabel, "0.0.0.0:8855",
+		model.JobLabel, "test",
+		model.MetricNameLabel, "",
+		"a", "b",
+		"a", "c",
+	)
+	_, err := tr.AppendExemplar(0, labels, exemplar.Exemplar{Value: 0})
+	require.Error(t, err)
+	assert.Contains(t, err.Error(), `invalid sample: non-unique label names: "a"`)
+}
+
+func TestAppendExemplarWithoutAddingMetric(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	labels := labels.FromStrings(
+		model.InstanceLabel, "0.0.0.0:8855",
+		model.JobLabel, "test",
+		model.MetricNameLabel, "counter_test",
+		"a", "b",
+	)
+	_, err := tr.AppendExemplar(0, labels, exemplar.Exemplar{Value: 0})
+	assert.NoError(t, err)
+}
+
+func TestAppendExemplarWithNoLabels(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	_, err := tr.AppendExemplar(0, nil, exemplar.Exemplar{Value: 0})
+	assert.Equal(t, errNoJobInstance, err)
+}
+
+func TestAppendExemplarWithEmptyLabelArray(t *testing.T) {
+	sink := new(consumertest.MetricsSink)
+	tr := newTransaction(scrapeCtx, &startTimeAdjuster{startTime: startTimestamp}, sink, nil, componenttest.NewNopReceiverCreateSettings(), nopObsRecv())
+
+	_, err := tr.AppendExemplar(0, []labels.Label{}, exemplar.Exemplar{Value: 0})
+	assert.Equal(t, errNoJobInstance, err)
+}
+
```

Comment:
Add a test for large/short trace Id since I don't see any.

Reply:
Added multiple testcases in `createDataPoint` for each metric type. That covers short/long/empty traceid and spanid